### DISCUSSION TITLE
[V5] Route

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#2584] [Help] [answered]
Hi team,
I want to use some route using authentication and without authentication both method. How this possible.
I mean I have a route /getPost I want to enable this route for guest user and login user both, When login use hit this route then auth.user value get.

@thetutlage

**Comment by RomainLanz** [comment, marked as answer]
Hey @pk4all! 👋 

You should use the `SilentAuth` middleware.
▶️ https://docs.adonisjs.com/guides/auth/middleware#silent-auth-middleware

**Reply by pk4all** [reply]
@RomainLanz Thanks
